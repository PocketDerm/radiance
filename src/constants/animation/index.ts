export default {
  defaultTiming: '350ms',
};
