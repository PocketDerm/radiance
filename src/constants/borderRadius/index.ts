const BORDER_RADIUS = {
  small: '4px',
  medium: '8px',
  large: '32px',
} as const;

export default BORDER_RADIUS;
