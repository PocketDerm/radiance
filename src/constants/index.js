export { default as ANIMATION } from './animation';
export { default as BOX_SHADOWS } from './boxShadows';
export { default as BREAKPOINTS } from './breakpoints';
export { default as COLORS } from './colors';
export { default as FONTS } from './fonts';
export { default as GRID } from './grid';
export { default as LAYOUT } from './layout';
export { default as MEDIA_QUERIES } from './mediaQueries';
export { default as PROGRESS_BAR_STATUS } from './progressBarStatus';
export { default as SPACING } from './spacing';
export { default as SPACER } from './spacer';
export { default as TYPOGRAPHY_CONSTANTS } from './typography';
export { default as Z_SCALE } from './zScale';
