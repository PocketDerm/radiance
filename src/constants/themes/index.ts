export { primaryTheme } from './primaryTheme';
export { secondaryTheme } from './secondaryTheme';
export * from './types';
