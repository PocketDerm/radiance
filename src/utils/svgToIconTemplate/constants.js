const UTIL_LOCATION = '../../../utils/icons';

module.exports = {
  UTIL_LOCATION,
};
