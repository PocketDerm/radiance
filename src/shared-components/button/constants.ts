import { COLORS } from '../../constants';

// TODO: potentially break out these pairings to the color constants file
export const textColorsAssociatedWithColors = {
  [COLORS.primary]: {
    tint1: COLORS.primaryTint1,
    tint2: COLORS.primaryTint2,
  },
  [COLORS.secondary]: {
    tint1: COLORS.lavender80,
    tint2: COLORS.lavender60,
  },
};
