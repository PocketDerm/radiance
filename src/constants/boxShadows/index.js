import { transparentize } from 'src/utils';

const boxShadowColor = '#2d2d30';
const boxShadowOverlayColor = '#505050';

const BASE_CONFIG = {
  clickable: `0 1px 3px 0 ${transparentize(boxShadowColor, 0.06)}`,
  clickableHover: `0 1px 3px 0 ${transparentize(boxShadowColor, 0.15)}`,
  message: `0 12px 20px 0 ${transparentize(boxShadowColor, 0.05)}`,
};

const OLD_BASE_CONFIG = {
  boxShadow1: `0 12px 20px 0 ${transparentize(boxShadowColor, 0.05)}`,
  boxShadow2: `0 5px 11px 0 ${transparentize(boxShadowColor, 0.1)}`,
  boxShadow3: `0 4px 6px 0 ${transparentize(boxShadowColor, 0.15)}`,
  boxShadow4: `0 1px 3px 0 ${transparentize(boxShadowColor, 0.15)}`,
  boxShadow5: `0 1px 2px 0 ${transparentize(boxShadowColor, 0.06)}`,
  boxShadowOnOverlay: `0 12px 20px 0 ${boxShadowOverlayColor}`,
};

const DESIGN_SYSTEM_NAMING = {
  elevation2: OLD_BASE_CONFIG.boxShadow5,
  elevation3: OLD_BASE_CONFIG.boxShadow4,
  elevation6: OLD_BASE_CONFIG.boxShadow3,
  elevation11: OLD_BASE_CONFIG.boxShadow2,
  elevation20: OLD_BASE_CONFIG.boxShadow1,
};

const ALIASED_CONFIG = {
  card: OLD_BASE_CONFIG.boxShadow4,
};

export default {
  ...BASE_CONFIG,
  ...OLD_BASE_CONFIG,
  ...DESIGN_SYSTEM_NAMING,
  ...ALIASED_CONFIG,
};

