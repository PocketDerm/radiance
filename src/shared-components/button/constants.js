import { COLORS } from 'src/constants';

//TODO: potentially break out these pairings to the color constants file
export const textColorsAssociatedWithColors = {
    purple: {
    	tint1: COLORS.purple85,
    	tint2: COLORS.purple70,
    },
    primary: {
    	tint1: COLORS.purple85,
    	tint2: COLORS.purple70,
    },
    purple100: {
    	tint1: COLORS.purple85,
    	tint2: COLORS.purple70,
    },
    lavender100: {
    	tint1: COLORS.lavender80,
    	tint2: COLORS.lavender60,
    },
    secondary: {
    	tint1: COLORS.lavender80,
    	tint2: COLORS.lavender60,
    },
};
