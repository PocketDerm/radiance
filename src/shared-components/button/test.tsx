import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { decorateWithThemeProvider } from 'src/tests/decorateWithThemeProvider';
import { primaryTheme } from 'src/constants/themes';

import CameraIcon from '../../svgs/icons/camera-icon.svg';

import { Button } from './index';

describe('<Button />', () => {
  const DecoratedButton = decorateWithThemeProvider(Button);

  describe('UI snapshots', () => {
    it('renders with props', () => {
      const tree = renderer
        .create(
          <DecoratedButton
            disabled
            onClick={() => undefined}
            icon={<CameraIcon />}
          >
            Button Text
          </DecoratedButton>,
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('renders with adjustable color', () => {
      const button = mount(
        <DecoratedButton
          buttonColor={primaryTheme.COLORS.error}
          onClick={() => undefined}
          id="red-button"
        >
          Button Text
        </DecoratedButton>,
      );

      expect(button.prop('buttonColor')).toEqual(primaryTheme.COLORS.error);
    });
  });

  describe('onClick callback', () => {
    it('should be invoked onClick', () => {
      const spy = jest.fn();
      const button = mount(
        <DecoratedButton onClick={spy}>Button Text</DecoratedButton>,
      );

      button.simulate('click');
      expect(spy).toHaveBeenCalled();
    });

    it('should not be invoked if disabled', () => {
      const spy = jest.fn();
      const button = mount(
        <DecoratedButton disabled onClick={spy}>
          Button Text
        </DecoratedButton>,
      );

      button.simulate('click');
      expect(spy).not.toHaveBeenCalled();
    });

    it('should not be invoked if loading', () => {
      const spy = jest.fn();
      const button = mount(
        <DecoratedButton isLoading onClick={spy}>
          Button Text
        </DecoratedButton>,
      );

      button.simulate('click');
      expect(spy).not.toHaveBeenCalled();
    });
  });
});
