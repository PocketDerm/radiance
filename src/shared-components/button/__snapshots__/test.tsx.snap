// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> UI snapshots renders with props 1`] = `
@keyframes animation-0 {
  0% {
    opacity: 1;
    -webkit-transform: translate3d(0,0,0) scale(1,1);
    -ms-transform: translate3d(0,0,0) scale(1,1);
    transform: translate3d(0,0,0) scale(1,1);
  }

  20% {
    opacity: .95;
    -webkit-transform: translate3d(12px,0,0) scale(1,1);
    -ms-transform: translate3d(12px,0,0) scale(1,1);
    transform: translate3d(12px,0,0) scale(1,1);
  }

  35% {
    opacity: 0;
    -webkit-transform: translate3d(16px,0,0) scale(.85,.85);
    -ms-transform: translate3d(16px,0,0) scale(.85,.85);
    transform: translate3d(16px,0,0) scale(.85,.85);
  }

  100% {
    opacity: 0;
    -webkit-transform: translate3d(16px,0,0);
    -ms-transform: translate3d(16px,0,0);
    transform: translate3d(16px,0,0);
  }
}

@keyframes animation-0 {
  0% {
    opacity: 1;
    -webkit-transform: translate3d(0,0,0) scale(1,1);
    -ms-transform: translate3d(0,0,0) scale(1,1);
    transform: translate3d(0,0,0) scale(1,1);
  }

  20% {
    opacity: .95;
    -webkit-transform: translate3d(12px,0,0) scale(1,1);
    -ms-transform: translate3d(12px,0,0) scale(1,1);
    transform: translate3d(12px,0,0) scale(1,1);
  }

  35% {
    opacity: 0;
    -webkit-transform: translate3d(16px,0,0) scale(.85,.85);
    -ms-transform: translate3d(16px,0,0) scale(.85,.85);
    transform: translate3d(16px,0,0) scale(.85,.85);
  }

  100% {
    opacity: 0;
    -webkit-transform: translate3d(16px,0,0);
    -ms-transform: translate3d(16px,0,0);
    transform: translate3d(16px,0,0);
  }
}

@keyframes animation-0 {
  0% {
    opacity: 1;
    -webkit-transform: translate3d(0,0,0) scale(1,1);
    -ms-transform: translate3d(0,0,0) scale(1,1);
    transform: translate3d(0,0,0) scale(1,1);
  }

  20% {
    opacity: .95;
    -webkit-transform: translate3d(12px,0,0) scale(1,1);
    -ms-transform: translate3d(12px,0,0) scale(1,1);
    transform: translate3d(12px,0,0) scale(1,1);
  }

  35% {
    opacity: 0;
    -webkit-transform: translate3d(16px,0,0) scale(.85,.85);
    -ms-transform: translate3d(16px,0,0) scale(.85,.85);
    transform: translate3d(16px,0,0) scale(.85,.85);
  }

  100% {
    opacity: 0;
    -webkit-transform: translate3d(16px,0,0);
    -ms-transform: translate3d(16px,0,0);
    transform: translate3d(16px,0,0);
  }
}

.emotion-7 {
  color: #524D6E;
  font-size: 0.75rem;
  line-height: 1.67;
  font-weight: bold;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  text-transform: uppercase;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border-radius: 0.25rem;
  border-style: solid;
  border-width: 1px;
  cursor: pointer;
  display: block;
  margin: 0;
  min-height: 52px;
  opacity: 1;
  padding: 0 1.5rem;
  position: relative;
  -webkit-transition: all 350ms ease-in-out;
  transition: all 350ms ease-in-out;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #ededf0;
  border-color: #ededf0;
  color: #c3c0cd;
  cursor: not-allowed;
  fill: #c3c0cd;
  min-width: 208px;
  max-width: 325px;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
}

.emotion-7:hover {
  -webkit-transition: all 350ms ease-in-out;
  transition: all 350ms ease-in-out;
}

.emotion-7:active,
.emotion-7:focus {
  outline: none;
  box-shadow: 0px 0px 0px 2px #ffffff,0px 0px 0px 4px #332e54;
}

.emotion-7:visited,
.emotion-7:hover {
  opacity: 1;
  color: #c3c0cd;
}

.emotion-3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 1rem 0;
  -webkit-transition: -webkit-transform 350ms;
  -webkit-transition: transform 350ms;
  transition: transform 350ms;
  width: 100%;
  -webkit-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
}

.emotion-3 > svg {
  opacity: 1;
  -webkit-transition: opacity 350ms;
  transition: opacity 350ms;
  margin-right: 1rem;
  margin-top: -5px;
}

.emotion-0 {
  display: block;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: color 350ms,-webkit-transform 350ms;
  -webkit-transition: color 350ms,transform 350ms;
  transition: color 350ms,transform 350ms;
}

.emotion-1 {
  line-height: 1.5;
  margin: 0;
  padding-top: 2px;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  position: absolute;
  bottom: 0;
  right: 0;
  top: 0;
  margin-top: -6px;
  width: 38px;
  opacity: 0;
}

.emotion-5 span {
  border-radius: 50%;
  display: inline-block;
  height: 4px;
  width: 4px;
  position: absolute;
}

.emotion-5 span:nth-of-type(1) {
  -webkit-animation: animation-0 1750ms infinite linear;
  animation: animation-0 1750ms infinite linear;
}

.emotion-5 span:nth-of-type(2) {
  -webkit-animation: animation-0 1750ms infinite linear -300ms;
  animation: animation-0 1750ms infinite linear -300ms;
}

.emotion-5 span:nth-of-type(3) {
  -webkit-animation: animation-0 1750ms infinite linear -600ms;
  animation: animation-0 1750ms infinite linear -600ms;
}

<button
  className="emotion-7 emotion-8"
  disabled={true}
  onClick={[Function]}
  type="button"
>
  <div
    className="emotion-3 emotion-4"
  >
    <svg
      className="emotion-0"
      height={16}
      width={16}
    />
    <span
      className="emotion-1 emotion-2"
    >
      Button Text
    </span>
  </div>
  <div
    className=" emotion-5 emotion-6"
    disabled={true}
  >
    <div>
      <span />
      <span />
      <span />
    </div>
  </div>
</button>
`;
